internal/servers/worker/internal/metric/proxy_ws_test.go

```diff
@@ -0,0 +1,371 @@
+package metric
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInitializeWebsocketCollectors(t *testing.T) {
+	// Assert that calling the function with a nil Registerer is a valid operation.
+	require.NotPanics(t, func() { InitializeWebsocketCollectors(nil) })
+
+	// Try with an actual Registerer.
+	r := prometheus.NewRegistry()
+	require.NotPanics(t, func() { InitializeWebsocketCollectors(r) })
+
+	// Expect that some metrics exist.
+	f, err := r.Gather()
+	require.NoError(t, err)
+	require.Greater(t, len(f), 0)
+}
+
+func TestHandlerWrapperWithHijacker(t *testing.T) {
+	h := http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) { // Mock for the proxy handler.
+		require.NotNil(t, req)
+		require.NotNil(t, rw)
+
+		// We only wrap the response writer in metrics code if it implements Hijacker.
+		require.IsType(t, &hijackerWrapper{}, rw)
+
+		mw := rw.(*hijackerWrapper)
+		require.NotNil(t, mw.Hijacker)
+		require.NotNil(t, mw.ResponseWriter)
+	})
+
+	rw := &testHttpHijacker{ResponseWriter: httptest.NewRecorder(), t: t}
+	req := httptest.NewRequest("GET", "/v1/proxy", nil)
+	wrapped := InstrumentWebsocketWrapper(h)
+	wrapped.ServeHTTP(rw, req)
```

Comment:
Please also assert that the request reaches `h`.  Here and the other `TestHandlerWrapper` test.

Reply:
Not sure I understood this - We [check if the request is not nil inside `h`](https://github.com/hashicorp/boundary/pull/1963/files/d75c3db7450006ce4c69bd97933efa1e1655688a#diff-efaa445d723714b9a9380282573e7f82a3cde48812571f9ba9aada7b189c86fdR34) - Did you have something different in mind here?

Reply:
Ah, I see what you mean now - We don't assert that `h` was called in the first place. Good point!

Reply:
Addressed [7b9e229](https://github.com/hashicorp/boundary/pull/1963/commits/7b9e229dbf34e3c573d4bc39f4e3d67872402cc4)